### ISSUE TITLE
Conv1D with dilation with "causal" padding isn't reducing the size of the sequence

### ISSUE THREAD

**Issue by chausies** [fchollet/keras#8751]
If you run

```
x = Input(shape=(seq_length, dim))
y = Conv1D(num_filts, 2, dilation_rate=8, padding="causal")(x)
print(y.shape)
```

Then the shape printed out will be `(?, seq_length, dim)`, which means the sequence length wasn't changed at all. But isn't this incorrect? Because the dilation makes the effective kernel width of the convolution 9, shouldn't the sequence length of the output be decremented by 8 because the causal kernel needs 9 values to perform its dot product and so doesn't output on the first 8 samples in the sequence?

**Comment by gabrieldemarmiesse** [comment]
Thank you for the detailed issue. I'll look into it.

**Comment by saftacatalinmihai** [comment]
any news on this ?

**Comment by emerrf** [comment]
I was looking into this today by chance and comparing it with the WaveNet implementation from:
https://github.com/munich-ai-labs/keras2-wavenet/blob/master/wavenet_utils.py#L29
I have the impression that the Munich AI labs developers had the same issue and they overridden `compute_output_shape`. Maybe @imdatsolak has more details. HTH

**Comment by gabrieldemarmiesse** [comment]
Their implementation is the same as the build in causal keras implementation judging from the code. It seems they were just not aware that `causal` already existed in keras.

**Comment by abderrahim** [comment]
I think causal should not be a value of padding, but a separate option that can be used with both padding='valid' and padding='same' (and padding='full' I guess, I don't know this option). 

What keras does now with padding='causal' is causal convolution with padding='same' and what @chausies is asking for is causal convolution with padding='valid'.

I propose adding a new causal option, and making padding='causal' a deprecated equivalent to causal=True, padding='same'. If there is no objection to this, I'll try to implement it.

**Comment by gabrieldemarmiesse** [comment]
Can you give a minimal example of `Conv1D(causal=True, padding='valid')` in the way you think about it? What would be the difference with `Conv1D(causal=False, padding='valid')`? Can you give an example with NumPy arrays so that it's explicit?

**Comment by abderrahim** [comment]
@gabrieldemarmiesse thanks for making me think more deeply about this. You are indeed right. It is a documentation issue.

**Comment by gabrieldemarmiesse** [comment]
Yould you have the time to do a PR on the docs to mention the zero padding?

**Comment by rjpg** [comment]
hello,

I am trying to understand the temporal relation with 'causal' padding ... 

My goal is to implement in 2D ... for Conv2D assuming the x-axis is the "time steps" and the y-axis are several variables...

for example,assuming K.set_image_dim_ordering("th"),  the input shape of CNN2D is (batch,channels, x, y) 
 
and we have (none, 1 , 128,9) means that we have 128 time steps and 9 variables in a multivariate time series problem. 

How hard it would be to create or include (with a new extended conv2D) the 'causal' padding in conv2D ? 

we could also choose if the "time dimension" is vertical in x or horizontal in y...
